src/charm.py

```diff
@@ -175,6 +187,29 @@ def _on_orchestrator_available(self, event: OrchestratorAvailableEvent):
             return
         self.unit.status = ActiveStatus()
 
+    def _on_lte_core_relation_joined(self, event: RelationJoinedEvent):
+        """Triggered when lte-core relation is joined.
+
+        AGW will provide the IP address of gtp_br0 interface if that is available.
+
+        Args:
+            event: Juju event (RelationJoinedEvent)
+
+        Returns:
+            None
+        """
+        try:
+            ip = netifaces.ifaddresses("gtp_br0")[netifaces.AF_INET][0]["addr"]
+            if not self._is_valid_ipv4_gateway(ip):
+                logger.warning("IP address of gtp_br0 not valid.")
+                event.defer()
+                return
```

Comment:
`set_core_information` already does the validation of the IP address and raises `AddressValueError` if the validation fails. What do you think about replacing this check with catching the `AddressValueError` excetion?

Reply:
Agreed and fixed.